app/Http/Controllers/Admin/ProjectCrudController.php

```diff
@@ -417,6 +426,26 @@ public function reAssess($id)
         return $project;
     }
 
+    public function store()
+    {
+        $this->calculateBudgetEur();
+
+        return $this->traitStore();
+    }
+
+    public function update()
+    {
+        $this->calculateBudgetEur();
+
+        return $this->traitUpdate();
+    }
+
+    public function calculateBudgetEur() {
+        $budget = $this->crud->getRequest()->budget;
+        $exchangeRate = $this->crud->getRequest()->exchange_rate;
+        $this->crud->getRequest()->request->set('budget_eur', $budget * $exchangeRate);
+    }
```

Comment:
The exchange rate given  in the request will be from the project's currency to the organisation's primary currency - which is not necessarily EUR.